src/posix/platform/system.c

```diff
@@ -144,8 +144,8 @@ otInstance *otSysInit(int aArgCount, char *aArgVector[])
     platformAlarmInit(speedUpFactor);
     platformRadioInit(radioFile, radioConfig);
     platformRandomInit();
-#if OPENTHREAD_ENABLE_PLATFORM_UDP
-    platformUdpInit();
+#if OPENTHREAD_ENABLE_PLATFORM_UDP && OPENTHREAD_ENABLE_PLATFORM_NETIF == 0
+    platformUdpInit(getenv("PLATFORM_NETIF"));
```

Comment:
What will happen if there is no PLATFORM_NETIF env?

Reply:
When `PLATFORM_NETIF` features is not enabled and user just want to use `PLATFORM_UDP`, user can set this environment, without it will result in segment fault. I think we can enhance by exiting with `OT_EXIT_INVALID_ARGUMENTS`.